generate_compose/odcs_requester.py

```diff
@@ -16,9 +19,21 @@ class ODCSRequestReference(ComposeReference):
 @dataclass(frozen=True)
 class ODCSRequester(ComposeRequester):
     """
-    Request a new ODCS compose based on compose configurations and return a reference
+    Request a new ODCS composes based on compose configurations and return a reference
     to the remote compose location.
     """
 
-    def __call__(self, configs: ComposeConfigurations) -> ODCSRequestReference:
-        raise NotImplementedError()
+    odcs: ODCS = ODCS("https://odcs.engineering.redhat.com/")
+
+    def __call__(self, compose_configs: ComposeConfigurations) -> ODCSRequestReferences:
+        assert isinstance(compose_configs, ODCSComposesConfigs)
```

Comment:
why do we need the type assertion? it break the abstraction of the protocol

Reply:
@gbenhaim what's the way around it? If we don't do this assertion, then the static checks will later on complain about accessing fields which are not part of the parent class.

Reply:
define a method/property on the protocol that provides you what you need.

Reply:
Can you elaborate? We want to be able to use a property of the sub class instance (`configs`) that is not a part of the protocol.